### PR TITLE
Optimize tagreader.yaml

### PR DESCRIPTION
**Pull request by jesserockz**
- remove the unneeded `globals` to use the template switches directly
- switch `on_loop` to `on_boot` as it was only ever run once

### CODE REVIEW

**Comment by magnusoverli** [comment]
Currently we are using an active version of the buzzer, making us unable to use the new rtttl component in ESPhome.
Switching to on_boot is a good idea!

**Comment by jesserockz** [comment]
Ah, I did not notice the active buzzer. Have reverted that part of the code

File: tagreader.yaml [hunk 1]
```diff
@@ -183,23 +131,24 @@ pn532:
             id: buzzer
             level: 50%
         - delay: 100ms
-        - output.turn_off: buzzer
-
-
-
-
+        - output.turn_off: buzzer    - if:
```
Comment (magnusoverli) [inline tagreader.yaml:134]: Fix invalid `- if:`

**Review by magnusoverli — CHANGES_REQUESTED** [review]

**Comment by jesserockz** [comment]
But also:
```
- output.esp8266_pwm.set_frequency:
    id: buzzer
    frequency: 500hz
- output.set_level:
    id: buzzer
    level: 50%
```

Is exactly what the `rtttl` component does per note...
```
output_->update_frequency(freq);
output_->set_level(0.5);
```
I don't have an active buzzer to try it, maybe it will work with the buzzer you are using?

**Comment by jesserockz** [comment]
If the current code is working to change the tone of the buzzer for the 3 beeps, then it must be a passive buzzer and not an active one. I don't believe active buzzers can change tone.
If it is active and just beeps the same every time then you just need a `switch` to turn the gpio pin on and off and not the `pwm` control.

**Comment by adonno** [comment]
> If the current code is working to change the tone of the buzzer for the 3 beeps, then it must be a passive buzzer and not an active one. I don't believe active buzzers can change tone.
> If it is active and just beeps the same every time then you just need a `switch` to turn the gpio pin on and off and not the `pwm` control.

I will try that out as soon as I get home. I hope it works, it would allow for a cleaner yaml

**Comment by jesserockz** [comment]
Awesome. Sleep time for me, but I pushed the `rtttl` version to another [branch](https://github.com/jesserockz/tagreader/tree/patch-2)

**Review by magnusoverli — APPROVED** [review]

**Pull request merged**